pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -124,22 +125,26 @@ type ClusterBuilder struct {
 // NewClusterBuilder builds an instance of ClusterBuilder.
 func NewClusterBuilder(proxy *model.Proxy, req *model.PushRequest, cache model.XdsCache) *ClusterBuilder {
 	cb := &ClusterBuilder{
-		serviceInstances:  proxy.ServiceInstances,
-		proxyID:           proxy.ID,
-		proxyType:         proxy.Type,
-		proxyVersion:      proxy.Metadata.IstioVersion,
-		sidecarScope:      proxy.SidecarScope,
-		passThroughBindIP: getPassthroughBindIP(proxy),
-		supportsIPv4:      proxy.SupportsIPv4(),
-		supportsIPv6:      proxy.SupportsIPv6(),
-		hbone:             proxy.EnableHBONE(),
-		locality:          proxy.Locality,
-		proxyLabels:       proxy.Labels,
-		proxyView:         proxy.GetView(),
-		proxyIPAddresses:  proxy.IPAddresses,
-		configNamespace:   proxy.ConfigNamespace,
-		req:               req,
-		cache:             cache,
+		serviceInstances: proxy.ServiceInstances,
+		proxyID:          proxy.ID,
+		proxyType:        proxy.Type,
+		proxyVersion:     proxy.Metadata.IstioVersion,
+		sidecarScope:     proxy.SidecarScope,
+		supportsIPv4:     proxy.SupportsIPv4(),
+		supportsIPv6:     proxy.SupportsIPv6(),
+		hbone:            proxy.EnableHBONE(),
+		locality:         proxy.Locality,
+		proxyLabels:      proxy.Labels,
+		proxyView:        proxy.GetView(),
+		proxyIPAddresses: proxy.IPAddresses,
+		configNamespace:  proxy.ConfigNamespace,
+		req:              req,
+		cache:            cache,
+	}
+	passthrougBindIPs := getPassthroughBindIPsForDualStack(proxy.GetIPMode())
```

Comment:
Do you anticipate >2 IPs? What if you do

```suggestion
	ipv4Bind, ipv6Bind := getPassthroughBindIPs(proxy.GetIPMode())
```

and that way can make IPv4 the first bind and IPv6 in the extra binds for dual stack.

Reply:
Hi @jacob-delgado, as you know, there are 2 kinds of IP family policies, so it's okay even if there are more than 2 IP addresses because each IP address should be included an IPv4 or IPv6 address.